src/Contract/Operation/Whenable.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/**
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace loophp\collection\Contract\Operation;
+
+use Iterator;
+use loophp\collection\Contract\Collection;
+
+/**
+ * @template TKey
+ * @template T
+ */
+interface Whenable
+{
+    /**
+     * @param callable(Iterator<TKey, T>): bool $predicate
+     * @param callable(Iterator<TKey, T>): iterable<TKey, T> $whenTrue
+     * @param callable(Iterator<TKey, T>): iterable<TKey, T> $whenFalse
+     *
+     * @return \loophp\collection\Collection<TKey, T>
+     */
+    public function when(callable $predicate, callable $whenTrue, ?callable $whenFalse = null): Collection;
```

Comment:
should this return `Collection` the implementation or `Collection` the contract?

Isn't there a mismatch here? the `@return` is showing the implementation, but the actual return type of the function is the contract